### PR TITLE
SHIFT SIGN UP list enchancement: show only not empty events and jobs

### LINKED ISSUE #175: SHIFT SIGN UP list enchancement: show only not empty events and jobs
In the "SHIFT SIGN UP" list we can see all jobs and events. Even events and jobs without shifts. Show only not empty events and jobs.

image

### PR DESCRIPTION
**Pull request by ahmedsabie**
EDIT: look at pull #185 instead
closes #175 
I made changes to:
/event/services.py
/event/views.py
/job/services.py
/job/views.py

The shift sign up will no longer show events with no jobs, or events with jobs but no shifts, or jobs with no shifts.

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
@ahmedsabie You should explain what a particular commit is doing in the commit message. *First commit* will not tell much when someone wants to understand what this commit is doing, how much it is affecting the code.

[These](http://chris.beams.io/posts/git-commit/) [two](https://github.com/erlang/otp/wiki/Writing-good-commit-messages) articles can help you. Looking at the previous commit will also help, see [here](https://github.com/systers/vms/commits/develop?page=2).

[Some other commits](https://gyazo.com/9335cf4fa9507f331669a79480bfbd23) are also coming in your PR, can you try fixing them as well.

File: vms/shift/services.py [hunk 1]
```diff
@@ -326,15 +326,15 @@ def get_shifts_with_open_slots(j_id):
...
     """
     shift_list_by_date = get_shifts_ordered_by_date(j_id)
     shift_list = []
-
+    current_date = date.today()
     for shift in shift_list_by_date:
         slots_remaining = get_shift_slots_remaining(shift.id)
-        if slots_remaining > 0 and not is_signed_up(v_id, shift.id):
+        if slots_remaining > 0 and not is_signed_up(v_id, shift.id) and shift.date >= current_date:
             shift_map = {}
```
Comment (tapasweni-pathak) [inline vms/shift/services.py]: This change should go in a separate pull request, fixing issue #130.

One pull request should have diffs, only concentrated for one fix or enhancement.

**Comment by tapasweni-pathak** [comment]
@ahmedsabie Please claim task VMS-22 in GCI portal as you have submitted a PR for the same.

**Comment by ahmedsabie** [comment]
@tapasweni-pathak I was going to fix them yesterday but the problem is the develop repository changed in the mean time. Should I redo changes with the new repository? Also I can't claim two tasks at the same time, so I'll claim it as soon as my current one is done.

**Comment by tapasweni-pathak** [comment]
@ahmedsabie You need to [update your fork](http://stackoverflow.com/questions/7244321/how-to-update-a-github-forked-repository), rebase and then update your pull request.


> Also I can't claim two tasks at the same time, so I'll claim it as soon as my current one is done.

Sure.

**Comment by tapasweni-pathak** [comment]
@ahmedsabie  As you have opened a new PR #185, so is it fine if I close this?

**Comment by ahmedsabie** [comment]
@tapasweni-pathak yea no problem! I just kept it as it had a reference to the new one.
